WordPress/Sniffs/DB/PreparedSQLPlaceholdersSniff.php

```diff
@@ -409,14 +409,30 @@ public function process_token( $stackPtr ) {
 				unset( $match, $matches );
 			}
 
+			/*
+			 * Analyse the query for quoted identifier placeholders.
+			 */
+			$regex = '/`%i`/';
+			if ( preg_match_all( $regex, $content, $matches ) > 0 ) {
```

Comment:
Thanks, yes, I was being lazy and cloned/tweaked the sniff above it.

Because I need to check Simple and Complex Identifier placeholders, I've re-written this.

Just to note, I am only looking for a quote character before - it keeps the RegEx simpler, and I can't think of a quote only appearing immediately after a backtick quoted identifier doing anything other than causing an SQL parse error.

Reply:
> Because I need to check Simple and Complex Identifier placeholders, I've re-written this.

👍🏻 

> Just to note, I am only looking for a quote character before - it keeps the RegEx simpler, and I can't think of a quote only appearing immediately after a backtick quoted identifier doing anything other than causing an SQL parse error.

That actually has the side-effect of making the error message look unbalanced (not the missing backtick after the placeholder):
```
 243 | ERROR   | Complex placeholders used for Identifiers (%i) in the query string in $wpdb->prepare() are always quoted automagically. Found:
     |         | `%1$'x10.3i. (WordPress.DB.PreparedSQLPlaceholders.QuotedIdentifierPlaceholder)
```

I have left a suggestion inline on how to improve the regex/resulting match use.

Reply:
Thanks, I've used your suggestion (more details below).